From 06ecd18c01afece5fac3c0f57a46592ca143645a Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Mon, 24 Mar 2025 14:17:21 +0100
Subject: [PATCH] Fix build with gcc-15

:Release Notes:
Add missing include.

:Detailed Notes:
Fixes:

http://gecko.lge.com:8000/Errors/Details/1142193

In file included from src/runtime/Allocator.cpp:25:
./arm_compute/runtime/MemoryRegion.h:105:21: error: 'uint8_t' was not declared in this scope
  105 |     std::shared_ptr<uint8_t> _mem;
      |                     ^~~~~~~

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRR-5443] Create GPVB with Yocto 5.2 Walnascar
---
Upstream-Status: Pending

 arm_compute/runtime/MemoryRegion.h                              | 1 +
 .../NEON/kernels/arm_conv/pooling/pooling_implementation.hpp    | 1 +
 tests/framework/instruments/hwc_names.hpp                       | 2 ++
 3 files changed, 4 insertions(+)

diff --git a/arm_compute/runtime/MemoryRegion.h b/arm_compute/runtime/MemoryRegion.h
index 6408deceaa..b671433dd2 100644
--- a/arm_compute/runtime/MemoryRegion.h
+++ b/arm_compute/runtime/MemoryRegion.h
@@ -29,6 +29,7 @@
 #include "arm_compute/core/Error.h"
 
 #include <cstddef>
+#include <cstdint>
 
 namespace arm_compute
 {
diff --git a/src/core/NEON/kernels/arm_conv/pooling/pooling_implementation.hpp b/src/core/NEON/kernels/arm_conv/pooling/pooling_implementation.hpp
index 78320cec44..afea425c5a 100644
--- a/src/core/NEON/kernels/arm_conv/pooling/pooling_implementation.hpp
+++ b/src/core/NEON/kernels/arm_conv/pooling/pooling_implementation.hpp
@@ -26,6 +26,7 @@
 #include "pooling.hpp"
 
 #include <cstddef>
+#include <cstdint>
 #include <functional>
 #include <cstring>
 
diff --git a/tests/framework/instruments/hwc_names.hpp b/tests/framework/instruments/hwc_names.hpp
index c39f3bba7a..e9d947d200 100644
--- a/tests/framework/instruments/hwc_names.hpp
+++ b/tests/framework/instruments/hwc_names.hpp
@@ -26,6 +26,8 @@
 
 #ifndef DOXYGEN_SKIP_THIS
 
+#include <cstdint>
+
 namespace mali_userspace
 {
 enum MaliCounterBlockName
